### ISSUE TITLE
OwnCloud stops responding randomly

### ISSUE THREAD

**Issue by jkalousek** [owncloud/core#18644]
I'm running clean ownCloud 8.1.1 (without apps) on Virtual machine (VirtualBox) with Ubuntu 14.04.3 LTS.
Apache/2.4.7, PHP 5.5.9-1ubuntu4.11 (cli), mysql Ver 14.14 Distrib 5.5.44.

Config:
```
$CONFIG = array (
  'instanceid' => (redacted),
  'passwordsalt' => (redacted),
  'secret' => (redacted),
  'trusted_domains' =>
  array (
    0 => (redacted),
  ),
  'datadirectory' => '/media/CloudDATA/owncloud_data',
  'overwrite.cli.url' => (redacted),
  'dbtype' => 'mysql',
  'version' => '8.1.1.3',
  'dbname' => 'OwnCloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => (redacted),
  'dbpassword' => (redacted),
  'logtimezone' => 'UTC',
  'installed' => true,
  'maintenance' => false,
  'mail_domain' => (redacted),
  'mail_from_address' => (redacted),
  'mail_smtpmode' => 'smtp',
  'mail_smtpsecure' => 'ssl',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => (redacted),
  'mail_smtpport' => '465',
  'mail_smtpname' => (redacted),
  'mail_smtppassword' => (redacted),
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'filelocking.enabled' => 'true',
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' =>
  array (
        'host' => 'localhost',
        'port' => 6379,
        'timeout' => 0,
        'dbindex' => 0,
  ),
);

```
I'm having trouble with OwnCloud since I first installed it (version 8.1) and can't figure out what causing the problem.
Simply OwnCloud stops responding (CONNECTION TIMED OUT) in totally random (as far as I can tell) intervals (sometimes few hours after start, sometimes 30 seconds after start) I can’t connect via browser neither via WebDAV.
Apache continue running fine (status is running) and even if I do 
```
/etc/init.d/apache2 restart

```
(Server successfully restarts) OwnCloud still not responding and only thing that can bring him up is complete virtual machine restart or – weirdly if I connect thru SSH (Putty) to command line, Own cloud start being accessible for short period of time but eventually stops responding again. Sometimes It came up from dead randomly without any interaction with server.
There are no errors in apache or OwnCloud logs. Server himself running smoothly whole time responding to ping and everything.
It’s clean system with only OwnCloud and dependencies installed on it. Originally I thought that problem that causing it is php5-apcu so I remove and purge it and install redis but no luck with that.
Maybe worth mentioning that I’m using Cron and that it’s running without error even if OwnCloud is not responding.
I'm little bit stuck right now because my knowledge of OwnCloud/Linux run out as I tried everything that came up my mind but nothing seems to have any effect to this in any way.

**Comment by jkalousek** [comment]
Additionally it seems that when server is under some load it manage to stay alive – just now when I spotted that server start working again after few hours of being unreachable I start copying some big data with WebDAV which took around 20 minutes and right after it was done (approximately 2 minutes) server stop responding again and in another 5 minutes was back online and stay that way since (15 minutes).
I check multiple times that apache is still the only one listening on port 443 and nothing blocking him from being accessible.

**Comment by jkalousek** [comment]
So no advice what I can try after 9 days?

**Comment by Xenopathic** [comment]
A connection timed out error is a problem between the client and the web server, in establishing a connection. I've noticed similar issues with VirtualBox on certain platforms before, where the whole machine seemingly locks to incoming connections, but never figured out what caused it. Interestingly, on some hosts it worked fine, on others it would be every 10 minutes.

Unfortunately there's nothing much ownCloud can do :disappointed:

**Comment by jkalousek** [comment]
But OwnCloud is only webpage I tried to put on virtual machine/apache that stop responding, PHPMyAdmin for example running nonstop and I have no problem with connecting to it. And I also tried just simple index.html on same configuration as OC and again never have trouble with connection timeout. So there must be something with OwnCloud that causing it when everything else I put on the server is running flawlessly.

And as I say, I can access everything else on VM during OC connection timeout so VM itself isn’t locked for incoming connections.

**Comment by PVince81** [comment]
Please try again with 8.1.3.
There was a case with infinite loop with shares that was fixed there, which could explain the load.

**Comment by jkalousek** [comment]
Installed, after 2 hours OC stop responding again.

**Comment by PVince81** [comment]
How many users, shares, etc do you have ?

**Comment by PVince81** [comment]
are you using the Pictures/GAllery app with a lot of pictures ?

**Comment by jkalousek** [comment]
I have one user (me) with 6 shared files.
I have planned to start using OwnCloud instead of One Drive so I just started move files to OC when timeouts starts happening.
Soo far I have few Excel documents, around 5GB of photos (my guess is around 3 000 pictures) and some music (backup of iTunes library). But it’s just very small part of my library.

Only apps that I have enabled right now are: Activity, Deleted files. File Locking, Updater

But I did not view any of my pictures throw own cloud gallery app before when it was enabled (I use WebDAV if needed).

**Comment by jkalousek** [comment]
But you could be right that it has something to do with photos because, after I restart server so that I can access it and when I tried access one of my folder with around 1000 photos in it server immediately stop responding. But I can remember that sometimes I can view photos in this folder through WebDAV without problem.

**Comment by PVince81** [comment]
Hmm strange. If you are accessing the folder in the web UI it would create thumbnails for the listed files. But unless you scroll down, it would only create thumbnails for the currently viewed page.
This shouldn't cause such load issues.

**Comment by jkalousek** [comment]
Is there any log file that would help you identify problem in some way?

**Comment by PVince81** [comment]
You can try setting "loglevel" to 0 and check "data/owncloud.log" for clues about what it might be doing.
If the CPU usage monitor shows "ImageMagick" or something then it would be about thumbnail creation.

**Comment by PVince81** [comment]
Any more luck with 8.1.3 or 8.2 ?

**Comment by jkalousek** [comment]
Sorry for my late answer. I have currently very little time due to my work. But I would try update OC this weekend and let it run for a while to see if update resolve the problem.

**Comment by jkalousek** [comment]
(Ok I just done it now) Well it seems that I don’t have 8.2 available for upgrade yet.
After I start OC this time these errors appeared in owncloud.log:
{"reqId":"5O3iiNdBV9QRckL22Pb6","remoteAddr":"192.168.-.---","app":"PHP","message":"Array to string conversion at \/var\/www\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-10-22T07:31:34+00:00","method":"GET","url":"\/index.php\/settings\/admin"}
{"reqId":"cG6ysofJ+rsEplyv56Vg","remoteAddr":"192.168.-.---","app":"PHP","message":"Array to string conversion at \/var\/www\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-10-22T07:37:50+00:00","method":"GET","url":"\/index.php\/settings\/admin"}
{"reqId":"m3PTeNS4fGMpv14mV5Yh","remoteAddr":"192.168.-.---","app":"PHP","message":"Array to string conversion at \/var\/www\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-10-22T07:41:32+00:00","method":"GET","url":"\/index.php\/settings\/admin"}

But it doesn't cause server to stop...

**Comment by PVince81** [comment]
The array to string conversion issue should be solved already in 8.2.2.
You said the server doesn't crash again, so it seems the problem is solved.
